### PR TITLE
Adds load_balancer resource template

### PR DESCRIPTION
**Pull request by tschwarz01**
# [Issue-id](https://github.com/Azure/caf-terraform-landingzones/issues/ISSUE-ID-GOES-HERE)

## PR Checklist

---

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X ] My code follows the code style of this project.
- [ ] I ran lint checks locally prior to submission.
- [X ] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

Adds load_balancers to the available resource templates

## Does this introduce a breaking change

- [ ] YES
- [X ] NO

## Testing

      load_balancers:
        lb-vmss:
          name: lb-vmss
          sku: basic
          resource_group_key: integration
          backend_address_pool_name: vmss1
          frontend_ip_configurations:
            config1:
              name: config1
              public_ip_address_key: lb_pip1
          probes:
            probe1:
              resource_group_key: integration
              load_balancer_key: lb-vmss
              probe_name: rdp
              port: 3389
          lb_rules:
            rule1:
              resource_group_key: integration
              load_balancer_key: lb-vmss
              lb_rule_name: rule1
              protocol: tcp
              probe_id_key: probe1
              frontend_port: 3389
              backend_port: 3389
              frontend_ip_configuration_name: config1

### CODE REVIEW

File: templates/resources/load_balancers.tfvars.j2 [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        vnet_key    = "{{config.vnet_key}}"
+{% endif %}
+{% if config.subnet_key is defined %}
+        subnet_key    = "{{config.subnet_key}}"
+{% endif %}
+{% if config.lz_key is defined %}
+        lz_key     = "{{ config.lz_key }}"
+{% endif %}
+{% if config.zones is defined %}
+        zones     = "{{ config.zones }}"
```
Comment (LaurentLesle) [inline templates/resources/load_balancers.tfvars.j2:40]: zones is a list. So you need to adjust the code to get it back to a terraform HCL list

```
{% if disk_value.zones is defined %}
          zones = {{ disk_value.zones | map('string') | replace('None','[]') | replace('\'','\"') }}
{% endif %}
```

File: templates/resources/load_balancers.tfvars.j2 [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+{% endif %}
+      }
+{% endfor %}
+    }
+{% endif %}
+{% if lbs.tags is defined %}
+      tags                = {{ lbs.tags }}
```
Comment (LaurentLesle) [inline templates/resources/load_balancers.tfvars.j2:47]: Same add the for loop

**Review by LaurentLesle — CHANGES_REQUESTED** [review]

File: templates/resources/load_balancers.tfvars.j2 (hunk 2 above)
Comment (tschwarz01) [reply templates/resources/load_balancers.tfvars.j2]: Complete

**Comment by LaurentLesle** [comment]
load_balancers is replaced by lb

**Review by LaurentLesle — APPROVED** [review]

**Pull request merged**